### ISSUE TITLE
[FEATURE] Configure dedicated CPU for Longhorn V2

### ISSUE THREAD

**Issue by innobead** [harvester/harvester#7227]
**Is your feature request related to a problem? Please describe.**

Enable the following features of the v2 data engine introduced in Longhorn 1.8 and 1.10.

- CPU core configuration https://github.com/longhorn/longhorn/issues/8835
- Interruption mode https://github.com/longhorn/longhorn/issues/9834

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

cc @harvester/storage

**Comment by Vicente-Cheng** [comment]
Here is the related PR (https://github.com/harvester/harvester/pull/6922)
We can use it to build ISO for the integration test.

**Comment by innobead** [comment]
I changed this for CPU core configuration, and the remaining issues have been planned into the corresponding milestones. 

- https://github.com/harvester/harvester/issues/7523
- https://github.com/harvester/harvester/issues/7522
- https://github.com/harvester/harvester/issues/7521

cc @harvester/tpm

**Comment by innobead** [comment]
Changed the scope to include both CPU core configuration and interruption mode for Longhorn v2. 

cc @harvester/storage

**Comment by Vicente-Cheng** [comment]
We found some conflict between the CPU pinning feature and LHv2. 
We need more time to have a good integration between CPU pinning and this feature.

At v1.8.0, we will add a document to tell the user the limitations we have.
cc @innobead, @brandboat

**Comment by innobead** [comment]
Removed interruption mode because it's not ready on the Longhorn side. We just need to focus on GA features in Longhorn V2. 

cc @harvester/storage @derekbit @c3y1huang

**Comment by innobead** [comment]
Moved this to ICEBOX because it's blocked by the upstream issue at https://github.com/longhorn/longhorn/issues/13248. A workaround isn't necessary since this V2 integration is only in preview. We'll address this in Harvester 1.10 with Longhorn 1.13, and we plan to promote V2 to GA in that release. 

@brandboat @Vicente-Cheng @harvester/storage, let me know if you have any concerns?

**Comment by innobead** [comment]
@derekbit confirmed that https://github.com/longhorn/longhorn/issues/13248 is still planned for backporting to Longhorn 1.12.1, which will be used in Harvester 1.9. @brandboat I moved this back to 1.9.0 with low priority, since it still depends on the upstream status. If we can't catch up with this feature in 1.9.0, we can move it to 1.10.0.

**Comment by Vicente-Cheng** [comment]
Longhorn would support CPU Manager or Dedicated CPU, but it needs more time to design the whole UI/UX on the Harvester side (integrate with the VM CPU PINNING). Moved to the next milestone.
We will have the document and KB on v1.9.0

**Comment by brandboat** [comment]
> Longhorn would support CPU Manager or Dedicated CPU, but it needs more time to design the whole UI/UX on the Harvester side (integrate with the VM CPU PINNING). Moved to the next milestone.
We will have the document and KB on v1.9.0

Doc issue is here: https://github.com/harvester/harvester/issues/11056

**Comment by WebberHuang1118** [comment]
We will use CPU Manager (lh setting `data-engine-number-of-cpu-cores`) for im pods instead of CPU mask.
